### PR TITLE
More ASN.1/ECC refactoring

### PR DESCRIPTION
**Pull request by simo5**
Further simplify and consolidate code.

This PR is a dependency for #110 from which it was split off

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -69,7 +70,9 @@ sqlitedb = ["dep:rusqlite"]
 # these are always required, so easier to specify this way
 default = [ "aes", "hmac", "pbkdf2", "sqlitedb" ]
 
-standard = [ "ecdsa", "ec_montgomery", "eddsa", "ecdh", "hash", "hkdf", "rsa", "sp800_108", "sshkdf", "tlskdf"]
+ecc_all = ["ecdsa", "ec_montgomery", "eddsa", "ecdh"]
+
+standard = [ "ecc_all", "hash", "hkdf", "rsa", "sp800_108", "sshkdf", "tlskdf"]
 
 fips = [ "rusqlite/bundled", "ecdsa", "ecdh", "hash", "hkdf", "rsa", "sp800_108", "sshkdf", "tlskdf"]
```
Comment (Jakuje) [inline Cargo.toml:77]: I think we wanted to enable eddsa for FIPS too, didn't we?

File: src/tests/ecdh_vectors.rs [hunk 2]
```diff
@@ -36,12 +35,22 @@ enum EcdhParserState {
     StateData,
 }
 
+pub fn map_curve_name(curve: &str) -> Option<&'static str> {
+    match curve {
+        "P-224" => Some("prime224v1"),
```
[the remark this answers is not in the document]
Comment (simo5) [reply src/tests/ecdh_vectors.rs:40]: Oh I remember that, will add a short comment

File: src/ec/mod.rs [hunk 3]
```diff
@@ -61,119 +30,45 @@ pub fn ec_key_check_import(obj: &mut Object) -> Result<()> {
...
-        OID_ED448 => Ok(NAME_ED448),
-        OID_X25519 => Ok(NAME_X25519),
-        OID_X448 => Ok(NAME_X448),
+        EC_SECP256R1 => Ok(BITS_SECP256R1),
+        EC_SECP384R1 => Ok(BITS_SECP384R1),
+        EC_SECP521R1 => Ok(BITS_SECP521R1),
+        ED25519_OID => Ok(BITS_ED25519),
+        ED448_OID => Ok(BITS_ED448),
+        X25519_OID => Ok(BITS_X25519),
+        X448_OID => Ok(BITS_X448),
```
[the remark this answers is not in the document]
Comment (simo5) [reply src/ec/mod.rs:58]: No the signature is ED25519, X25519 is exactly key eXchange

The Xnnn_OID names come from the pyca files which I do not want to change, so we can easily swap them out once those become actual separate crates (already discussed with the pyca maintainers).

The CURVE name comes for standards, so this is the best compromise.

File: Cargo.toml (hunk 1 above)
Comment (simo5) [reply Cargo.toml:77]: Yes eventually, but to do that we need either a new version of OpenSSL, or we need to add selftests ourselves, so "not yet".

**Review by Jakuje — APPROVED** [review]
lgtm

**Pull request merged**